### DISCUSSION TITLE
Bazel Apple/Swift Ecosystem - 12/2024

### DISCUSSION THREAD

**Discussion by luispadron** [bazelbuild/rules_apple#2621] [General]
Thank you all for attending the first meeting! Was great to hear from you all and we had some great discussions. Continue any discussions in the #apple Slack channel or in GitHub!

Please take some time to fill [this information gathering form](https://forms.gle/r1he76FkL1umwKSu8) out so we can learn about our users more.

You can find the notes for the meeting here: https://docs.google.com/document/d/1r7ea1YInXV1exBtPskl5j6JIoUEaiijCKTuXbjeK1CE

**Comment by luispadron** [comment]
Hi again folks!

I wanted to post the results of our first information form we shared out at the start of the meeting. It's been a few days and most of the people that attended have now responded, the full results are available here: https://docs.google.com/forms/d/1_lBwfuLfYFdSS-VLyHZc0I0T0rJsmUq3YAvYSUFwRqc/edit#responses

Below is some of the more interesting data points.

## 100% of users are on Bazel 7

<img width="600" alt="Screenshot 2024-12-16 at 12 10 07 PM" src="https://github.com/user-attachments/assets/63650439-7d4e-4d0c-a093-04651c855dab" />

## Majority of users are on the latest rule versions

<img width="582" alt="Screenshot 2024-12-16 at 12 10 37 PM" src="https://github.com/user-attachments/assets/0e9eaaed-f225-415b-901c-8b5fe0c47bc8" />

<img width="589" alt="Screenshot 2024-12-16 at 12 10 43 PM" src="https://github.com/user-attachments/assets/fbe040bb-6472-468d-befd-76f92018870b" />

## rules_xcodeproj and rules_swift_package_manager used heavily across the community

<img width="738" alt="Screenshot 2024-12-16 at 12 11 12 PM" src="https://github.com/user-attachments/assets/84644cb2-f3d9-47fc-b666-65f566b0e941" />

## Test runner usage is very split between the different implementations

<img width="624" alt="Screenshot 2024-12-16 at 12 11 54 PM" src="https://github.com/user-attachments/assets/b5dc6955-c856-4878-89c0-cd29a95d0db1" />

## Most of us are building iOS applications

<img width="745" alt="Screenshot 2024-12-16 at 12 12 31 PM" src="https://github.com/user-attachments/assets/c9e4ec8f-b4d9-4581-b0f2-dcde1e22553b" />

## Majority of users have actively been using Bazel but we still have some new users!

<img width="744" alt="Screenshot 2024-12-16 at 12 12 54 PM" src="https://github.com/user-attachments/assets/42527484-7639-49ca-b350-3a2c0a376131" />

**Comment by brentleyjones** [comment]
The Iink to see the responses doesn’t work, it just says the form is closed.

**Reply by luispadron** [reply]
My bad, I'm not sure how to make the responses page public. I'll need to figure out permissions for next time. Members of the Google group should have access to this export: https://docs.google.com/spreadsheets/d/1_l30Y8gbGvF4sIKo3qzbqfVWblZ7mEqhlYZaqqiL9IU (still not public though since its tied to my work account, will figure it out for next meeting)

**Comment by ssarad** [comment]
I would be very curious and interesting to learn on some of these responses. For example, we have a very split decision on test runner choice. I'd love to learn what made you use custom test runner vs ios_test_runner vs ios_xctestrun_runner ?

**Reply by keith** [reply]
Hopefully it's just that the default is still `ios_test_runner`, so you have to opt into `ios_xctestrun_runner`. If anyone spots issues with the new one please let us know. The long term goal was to flip the defaults and eventually remove the old one. We should have flipped it already but just haven't gotten to it at the same time as doing a major version

**Reply by luispadron** [reply]
Yeah I think the default will help get most of the folks using `ios_test_runner`. For our case (Square/Cash App) we use a fork of `ios_test_runner` which supports concurrent shared simulator tests.

I [shared some performance benchmarks](https://bazelbuild.slack.com/archives/CD3QY5C2X/p1718993839377589) earlier this year in the Bazel slack and the biggest thing stopping us from moving to `ios_xctestrun_runner` is the fact that when `reuse_simulator = True` you are forced to use `--local_test_jobs=1` (or `tags = ["exclusive"]`) for hosted tests so we lose a good amount of time running a bunch of things 1 by 1. If we can fix this in the runner I think it would get the remaining people over to `rules_apple` again.

![overall_comparison](https://github.com/user-attachments/assets/8b51078a-4a8c-4a29-8376-5432a2588e7f)

**Reply by ssarad** [reply]
Thats awesome, thanks for sharing the graph @luispadron !
